editor/src/templates/editor-canvas.tsx

```diff
@@ -909,42 +915,70 @@ export class EditorCanvas extends React.Component<EditorCanvasProps> {
             return
           }
 
-          const newUID = generateUidWithExistingComponents(this.props.editor.projectContents)
+          const position = this.getPosition(event.nativeEvent)
 
-          const newElement = createJsxImage(newUID, {
-            width: 1,
-            height: 1,
-          })
+          const setDragSessionStateActions =
+            this.props.editor.dragSessionState.type !== 'DRAGGING_FROM_SIDEBAR'
+              ? [EditorActions.setDragSessionState(draggingFromFS())]
+              : []
 
-          const position = this.getPosition(event.nativeEvent)
+          const newUID = generateUidWithExistingComponents(this.props.editor.projectContents)
 
-          this.props.dispatch(
-            [
-              EditorActions.enableInsertModeForJSXElement(newElement, newUID, {}, null),
-              CanvasActions.createInteractionSession(
-                createInteractionViaMouse(
-                  position.canvasPositionRounded,
-                  emptyModifiers,
-                  boundingArea(),
-                ),
-              ),
-            ],
-            'everyone',
+          const newElementProps: Partial<DraggedImageProperties> =
+            this.props.editor.dragSessionState.type === 'DRAGGING_FROM_SIDEBAR'
```

Comment:
It looks like this condition should probably be used for an early return from the `onDragOver` event

Reply:
That would also mean the above `Partial<DraggedImageProperties>` can become `DraggedImageProperties`, meaning no need for the null fallbacks below

Reply:
discussed on DC: https://discord.com/channels/647102474519576589/1024328724268732476/1030136332971946027